### DISCUSSION TITLE
How can I distinguish between different streams

### DISCUSSION THREAD

**Discussion by AvatarGanymede** [microsoft/msquic#2590] [Q&A] [answered]
Suppose that I open 3 streams on the server side. Is it possible to distinguish between them on the client side **without adding extra info in the payload**?

**Comment by nibanks** [comment, marked as answer]
Yes, you can query the ID of each stream via [`QUIC_PARAM_STREAM_ID`](https://github.com/microsoft/msquic/blob/main/src/inc/msquic.h#L781).
```c
    QUIC_UINT62 ID;
    uint32_t IDLen = sizeof(ID);
    MsQuic->GetParam(
        StreamHandle,
        QUIC_PARAM_STREAM_ID,
        &IDLen,
        &ID);
```
It's also worth noting that you can only query a stream's ID **after** it has been started; meaning just calling `StreamOpen` is not enough. When you receive a stream from the peer, it is already started.
